src/dbms/handler.hpp

```diff
@@ -62,8 +62,8 @@ class Handler {
    * @param name Name associated with the wanted context
    * @return std::optional<std::shared_ptr<TContext>>
    */
-  std::optional<std::shared_ptr<TContext>> Get(const std::string &name) {
-    if (auto search = items_.find(name); search != items_.end()) {
+  std::optional<std::shared_ptr<TContext>> Get(std::string_view name) {
+    if (auto search = items_.find(name.data()); search != items_.end()) {
```

Comment:
Nope, not guaranteed to be null terminated c string.

Unfortunately its an unordered_map, so it will construct a string anyway...just to hash it.

If the container didn't have to hash we could do heterogeneous lookup, by using heterogeneous comparator like `std::less<>`.

Reply:
Ah, you're right :(
Didn't want to think about it. I'll fix it. There are other similar places

Reply:
So turns out ([P0919](https://wg21.link/P0919R3) & [P1690](https://wg21.link/P1690R1)) now possible in c++20

Reply:
How to implement:
https://www.cppstories.com/2021/heterogeneous-access-cpp20/

Reply:
Do we actually want to do this? I doubt there is a huge gain.